server/src/test/java/io/crate/integrationtests/LogicalReplicationITest.java

```diff
@@ -257,6 +260,35 @@ public void test_subscribing_to_unknown_publication_raises_error() throws Except
             .hasMessageContaining("Publication 'unknown_pub' unknown");
     }
 
+    @Test
+    @UseRandomizedSchema(random = false)
+    public void test_subscribing_to_tables_with_higher_version_raises_error() throws Exception {
+        // Table to be replicated has a higher version subsriber's minimal node version.
+        publisherCluster.client()
+            .admin()
+            .indices()
+            .create(
+                new CreateIndexRequest("test")
+                .settings(
+                    Settings.builder()
+                        // Not necessary to set a valid version, only fact that version is higher than CURRENT is important for this test.
+                        .put("index.version.created", Version.CURRENT.internalId + 1)
+                        .put("number_of_shards", 2) // must be supplied
+                        .put("number_of_replicas", 0).build()
+                )
+            )
+            .get();
+
+        createPublication("pub1", false, List.of("doc.test"));
+        assertThatThrownBy(() -> executeOnSubscriber("CREATE SUBSCRIPTION sub1" +
+            " CONNECTION '" + publisherConnectionUrl() + "' publication pub1"))
+            .isExactlyInstanceOf(IllegalStateException.class)
+            .hasMessageContaining("One of the published tables has version higher than subscriber's minimal node version.");
+
+        var response = executeOnSubscriber("SELECT * FROM pg_subscription");
+        assertThat(response).hasRowCount(0);
+    }
+
```

Comment:
One of 2 tests should be removed. I added an integration test as I also wanted to test that subscription is not created - or it's an obvious implication from unit test, showing that transport action throws an error?

Reply:
The transport action test should be enough. Testing that tables aren't created if the action throws any error is not related to this change.